utils/python/CIME/utils.py

```diff
@@ -0,0 +1,65 @@
+"""
+Common functions used by cime python scripts
+"""
+import logging
+import sys
+import os
```

Comment:
I can't comment on the Test class since I'm not sure what that is yet. I will, however, caution against carried away with the OO paradigm. Free functions are a very useful concept. Their simplicity makes them preferable to OO, in my opinion, unless there's a lot of persistent/shared state involved.